ledger-core/v2/virtual/execute/execute_test.go

```diff
@@ -276,3 +254,168 @@ func TestSMExecute_Semi_IncrementPendingCounters(t *testing.T) {
+			CallOutgoing: outgoing,
+
+			Caller:              caller,
+			Callee:              callee,
+			CallSiteDeclaration: prototype,
+			CallSiteMethod:      "New",
+		},
+		Meta: &payload.Meta{
+			Sender: caller,
+		},
+	}
+
+	{
+		catalogWrapper := object.NewCatalogMockWrapper(mc)
+		var catalog object.Catalog = catalogWrapper.Mock()
+		slotMachine.AddInterfaceDependency(&catalog)
+
+		sharedStateData := smachine.NewUnboundSharedData(sharedState)
+		smObjectAccessor := object.SharedStateAccessor{SharedDataLink: sharedStateData}
+
+		catalogWrapper.AddObject(objectRef, smObjectAccessor)
+	}
+
+	{
+		pd := pulse.NewFirstPulsarData(10, longbits.Bits256{})
+		pulseSlot := conveyor.NewPresentPulseSlot(nil, pd.AsRange())
+		slotMachine.AddDependency(&pulseSlot)
+	}
+
+	slotMachine.Start()
+	defer slotMachine.Stop()
+
+	smWrapper := slotMachine.AddStateMachine(ctx, &smExecute)
+
+	require.False(t, smExecute.migrationHappened)
+
+	stepToWait := smExecute.stepTakeLock
+	if !slotMachine.StepUntil(smWrapper.WaitStep(stepToWait)) {
+		t.FailNow()
+	}
+
+	slotMachine.Migrate()
+
+	if !slotMachine.StepUntil(smWrapper.WaitAnyMigrate()) {
+		t.Error("slotmachine stopped")
+		t.FailNow()
+	}
+
+	require.False(t, smExecute.migrationHappened)
+
+	mc.Finish()
+}
+
+func TestSMExecute_MigrateAfterLock(t *testing.T) {
+	var (
+		mc  = minimock.NewController(t)
+		ctx = inslogger.TestContext(t)
+
+		prototype   = gen.UniqueReference()
+		caller      = gen.UniqueReference()
+		callee      = gen.UniqueReference()
+		outgoing    = gen.UniqueID()
+		objectRef   = reference.NewSelf(outgoing)
+		sharedState = &object.SharedState{
+			Info: object.Info{
+				KnownRequests:  make(map[reference.Global]struct{}),
+				ReadyToWork:    smsync.NewConditional(1, "ReadyToWork").SyncLink(),
+				MutableExecute: smsync.NewConditional(1, "MutableExecution").SyncLink(),
+			},
+		}
+	)
+
+	slotMachine := slotmachine.NewControlledSlotMachine(ctx, t, true)
+	slotMachine.PrepareMockedMessageSender(mc)
+	slotMachine.PrepareRunner(mc)
+
+	smExecute := SMExecute{
+		Payload: &payload.VCallRequest{
+			Polymorph:    uint32(payload.TypeVCallRequest),
+			CallType:     payload.CTConstructor,
+			CallFlags:    payload.BuildCallRequestFlags(contract.CallTolerable, contract.CallDirty),
+			CallOutgoing: outgoing,
+
+			Caller:              caller,
+			Callee:              callee,
+			CallSiteDeclaration: prototype,
+			CallSiteMethod:      "New",
+		},
+		Meta: &payload.Meta{
+			Sender: caller,
+		},
+	}
+
+	{
+		catalogWrapper := object.NewCatalogMockWrapper(mc)
+		var catalog object.Catalog = catalogWrapper.Mock()
+		slotMachine.AddInterfaceDependency(&catalog)
+
+		sharedStateData := smachine.NewUnboundSharedData(sharedState)
+		smObjectAccessor := object.SharedStateAccessor{SharedDataLink: sharedStateData}
+
+		catalogWrapper.AddObject(objectRef, smObjectAccessor)
+	}
+
+	{
+		pd := pulse.NewFirstPulsarData(10, longbits.Bits256{})
```

Comment:
It's same part in every test, please extract it to `slotMachine.PreparePresentPulseSlot()`

Reply:
there is not slotMachine in these tests

Reply:
![image](https://user-images.githubusercontent.com/495636/82922035-c7ce3900-9f81-11ea-8349-758c08efaf81.png)

Reply:
there is no pd in "SemiIntegrational", there is no slotMachine in unit tests